### PR TITLE
Cc/loopy login

### PR DESCRIPTION
**Pull request by Thurst-ft**
couple of merge conflicts and bumped the version to 1.0.0.

### CODE REVIEW

File: lib/middleware/errors.js [hunk 1]
```diff
@@ -19,19 +20,25 @@ function _applicationError(err, req, res, next) {
   logger.log('error', {'Error stack': err.stack});
 
   next(err);
+
+  return Promise.resolve();
```
Comment (dapenguin) [inline lib/middleware/errors.js]: Don't think this line would be reached (or useful) as `next()` sends the router to the next bit of middleware.

File: lib/middleware/verifySession.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 const logger = require('./../logger');
+const config = require('../config');
 const uriConstructor = require('./../uriConstructor');
 const cookieHandler = require('./../helpers/cookieHandler');
 const sessionClient = require('kat-client-proxies').sessionClient;
 
-function getUserId(req, res, next) {
+function getUserId (req, res, next) {
```
Comment (dapenguin) [inline lib/middleware/verifySession.js]: Any reason for the space between the function name and the parenthesis?

File: lib/middleware/errors.js [hunk 3]
```diff
@@ -24,14 +25,18 @@ function _applicationError(err, req, res, next) {
 function _authenticationError(err, req, res, next) {
   logger.log('warn', {'Unauthorized': 'Invalid user credentials'});
   logger.log('warn', {'Error stack': err.stack});
-  res.redirect(uriConstructor.redirectUrl(req.originalUrl));
+
+  return fetch(config.LOGOUT_URL, { headers: req.headers })
```
Comment (ciprianlujeru) [inline lib/middleware/errors.js]: Should we handle the error here too? I mean the request might fail but we still want to get redirected to the login page.
So we can maybe have:
```
return fetch(config.LOGOUT_URL, { headers: req.headers })
    .catch(err => err)// or null, or w/e
    .then(() => {
      res.redirect(uriConstructor.redirectUrl(req));
    });
```

File: lib/middleware/redirectToExistingSession.js [hunk 4]
```diff
@@ -19,8 +24,14 @@ module.exports = (req, res, next) => {
                 logger.info({operation, redirect: 'to accounts.ft.com/login', reason: 'no FTSession found'});
 
                 req.session = null;
-                res.redirect(uriConstructor.redirectUrl(req.originalUrl));
-                return resolve();
+
+                return fetch(config.LOGOUT_URL, {headers: req.headers})
```
Comment (ciprianlujeru) [inline lib/middleware/redirectToExistingSession.js]: Same as the other one. We should probably handle the error.

File: lib/middleware/errors.js (hunk 3 above)
Comment (Thurst-ft) [inline lib/middleware/errors.js]: So just swallow the error?

Comment (ciprianlujeru) [inline lib/middleware/errors.js]: Yes, so that both when the request succeeds or fails, the redirect happens.

Comment (Thurst-ft) [inline lib/middleware/errors.js]: Looking at that again I'm not entirly sure I understand why the res.redirect needs to be wrapped in a fetch. I think just redirecting to the LOGOUT_URL would solve the issue. I'm reading that as make a request ot LOGOUT_URL then redirect to LOGIN_URL I'm fairly sure LOGOUT_URL takes the same location parameter as LOGIN_URL

Comment (ciprianlujeru) [inline lib/middleware/errors.js]: Might be so. I was just commenting on the code. But yes, you might be right. We might not need this.

Comment (Thurst-ft) [inline lib/middleware/errors.js]: @ciprianlujeru @dapenguin can one of you approve

**Pull request merged**